spec/support/vcr.rb

```diff
@@ -6,12 +6,16 @@
   config.cassette_library_dir = 'spec/fixtures/slack'
   config.hook_into :webmock
   config.configure_rspec_metadata!
+  config.filter_sensitive_data('<SLACK_API_TOKEN>') { ENV['SLACK_API_TOKEN'] }
   config.before_record do |i|
-    i.request.body.gsub!(ENV['SLACK_API_TOKEN'], 'token') if ENV.key?('SLACK_API_TOKEN')
     i.response.body.force_encoding('UTF-8')
   end
+  config.register_request_matcher :client_headers do |request1, request2|
```

Comment:
Using a custom matcher rather than the build in `headers` matcher means we can match only on the headers the slack API cares about and not worrying about slight version differences in User-Agent, for example, or other headers that Faraday may be setting by default. The downside is that any error messages reported by VCR about not finding a matching request won't include the headers of the request, which makes diagnosing mismatches a bit problematic.

Reply:
I think I might be able to make this easier using another filter. I'll try it out when I am back at my desk.

Reply:
@dblock follow up PR here https://github.com/slack-ruby/slack-ruby-client/pull/392